### DISCUSSION TITLE
2008/09/12/%E7%9A%AE%E8%84%82%E8%85%BA%E5%9B%8A%E8%82%BF

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#243] [Announcements]
# 2008/09/12/%E7%9A%AE%E8%84%82%E8%85%BA%E5%9B%8A%E8%82%BF

最近跑医院比上班还勤，都是让背上一个皮脂腺囊肿害得。这个囊肿在我身上也有年头了。大约4年前的某天，我洗完澡闲着没事在后背上乱抓，突然摸到一个大包。平时后背也会长些类似小痘痘之类的小红包，但是这个包的个头要大很多。我赶紧对着镜子仔细研究起了这个包包。

https://blog.qizhen.xyz/2008/09/12/%E7%9A%AE%E8%84%82%E8%85%BA%E5%9B%8A%E8%82%BF

**Comment by ruanqizhen** [comment]
之前博客下的留言：

Qizhen
2008年10月9日 10:40 编辑
我的荨麻疹一直有，不过症状不重，所以我平时就当它不存在。

回复
斯
2008年10月8日 00:39 编辑
病友，你的荀麻疹还有再发没？

回复
Qizhen
2008年09月21日 19:30 编辑
如果是CPU使用率高，最可能就是程序中存在某些不必要的计算。这需要调试一下，先找出程序中最耗时的部分，然后针对其进行优化。我以前写的两篇文章，也许对你有帮助：LabVIEW 的运行效率 1 – 找到程序运行速度的瓶颈
LabVIEW 的运行效率 2 – 程序慢在哪里

回复
死了都要C
2008年09月19日 21:53 编辑
多谢阮大侠！！
6608测下降沿跳变的已经解决了，是数据库通道使用完未释放。
还是我那程序运行时，速度有点慢。
查看任务管理器内存占用并未增加，只是CPU使用率暴高，基本上都是平的，
而且有一个特点是把LV关掉重新开一次，速度能变快一些，但过一会就不行了。
现在老板急得让我交活，阮大侠有时间帮小弟看下时什么原因。

回复
Qizhen
2008年09月16日 15:09 编辑
死了都要C，不好意思，今天才来看你的问题。我只能回答你的第一个问题，后面与硬件相关的问题我不是很了解。程序越来越慢很可能是打开资源没释放引起的，除了动态调用，其它地方也可能会有资源泄漏。你可以打开系统的任务管理器观察一下程序占用的内存是不是越来越多，如果是，就说明有内存或者资源泄漏的问题。这主要就是打开了某些引用，没有关闭引起的。比如动态调用VI，每次打开的VI的引用都要被关闭才行。

回复
Xiang
2008年09月16日 09:04 编辑
呵呵，我8月份也做了引流手术。。。
等长好了还要去开一刀，而且这个要天天换药，麻烦的要死。。。

回复
死了都要C
2008年09月14日 12:26 编辑
阮大侠 保重身体 中秋快乐啊
能不能抽时间帮小弟看下问题
1.程序运行速度的问题。目前做一个电路板的测试，测试项有很多，40多个大项，大项中有30多个小项，
     程序的运行是采用动态调用实现的，刚开机运行速度还可以，但过一会非常慢，
     是否与动态调用的资源释放有关系？
   2.6608板卡测量两个信号跳变的滞后时间，
      基本原理是用LV例程中的TWO EDGE SEPERATION MEASURENT.VI来实现的，硬件上采用先到信号连至AUX端，后发信号连至GATE端。
      测量两个信号上升沿跳变的滞后时间是可行的，但测下降沿时始终是0，实在不知原因为何。
  3.6608测量两个信号跳变的滞后时间的时机问题。
     由于要测的这两个脚（PIN100.PIN101）信号的跳变是因另外一脚(PIN99)加高电平引起的，而这两个信号的跳变滞后时间非常短1MS左右，99脚加电平也是通过一数字板卡实现的（PXI6120），目前采取的方案是测量时间的VI跟99脚加电平并行执行，即将这两个VI的ERROR IN 并联。
      请教阮大侠有没有更好的方案。

回复
Unknown
2008年09月14日 10:41 编辑
首先祝阮大侠早日康复！
听说大侠要出书，谈点我对LabVIEW书籍的理解吧…
本人是LabVIEW新手，进入该行5个月左右，经历过一两个测控项目；初级的书读过：“LabVIEW8.20从入门到精通”、“LabVIEW For Eeryone”；现在正在读“The LabVIEW Style Book”。
纵观国内市场上关于LabVIEW的书籍，大多都是介绍一些很基础性的东西，从最早杨乐平系列，到“LabVIEW For Eeryone”（翻译版）无一例外；
叹自己的英文水平，国外一些高级教程如“Advanced ..”等实在是读不下去！
相对于初级入门者来说，市场上的书+LabVIEW自带帮助已经足够了！当一个新手把这些书看完，再经历几个项目，编程水平应该达到学习生涯的第一个瓶颈了.LabVIEW多用于工控领域，LabVIEW程序员也多半路出家，所以我认为这个瓶颈持续期要较长，可能会达到一年多。我相信国内大部分LabVIEW用户水平都在此处聚集！..以我以及我身边不少同学（学习时间和我差不多）的情况来说，目前最迫切看到的书应该包含以下：
1.设计模式项目实战；
2.数据流思想的讲解；
3.程序效率，如引用或局部变量怎么在内存中开辟空间等更贴近硬件的讲解.
4.高级应用；如Xcontrol等大侠博客中的一些文章都不错！
5.编程风格！
 
遇见仅供参考！
 
祝大侠早日成功！

回复
Peter
2008年09月14日 10:20 编辑
你身体的问题还真不少！！多运动运动吧恶心巴拉~~这个词好像南方人大多听不懂吧？

回复
WL
2008年09月13日 19:51 编辑
早日康复

回复
晖
2008年09月13日 13:36 编辑
原来囊肿这么恐怖。。。

回复
死了都要C
2008年09月13日 10:41 编辑
阮大侠：
  多注意身体呵~~
  小弟新用LV，目前遇到一些难题，希望阮大侠能帮组解决。  
   1.程序运行速度的问题。目前做一个电路板的测试，测试项有很多，40多个大项，大项中有30多个小项，
     程序的运行是采用动态调用实现的，刚开机运行速度还可以，但过一会非常慢，
     是否与动态调用的资源释放有关系？
   2.6608板卡测量两个信号跳变的滞后时间，
      基本原理是用LV例程中的TWO EDGE SEPERATION MEASURENT.VI来实现的，硬件上采用先到信号连至AUX端，后发信号连至GATE端。
      测量两个信号上升沿跳变的滞后时间是可行的，但测下降沿时始终是0，实在不知原因为何。
  3.6608测量两个信号跳变的滞后时间的时机问题。
     由于要测的这两个脚（PIN100.PIN101）信号的跳变是因另外一脚(PIN99)加高电平引起的，而这两个信号的跳变滞后时间非常短1MS左右，99脚加电平也是通过一数字板卡实现的（PXI6120），目前采取的方案是测量时间的VI跟99脚加电平并行执行，即将这两个VI的ERROR IN 并联。
      请教阮大侠有没有更好的方案。
       不胜感激。

回复
tang
2008年09月12日 16:52 编辑
注意身体！身体最重要

回复
Wu
2008年09月12日 16:17 编辑
受罪啊。。。

回复
Qizhen
2008年09月12日 15:49 编辑
感染了，处理起来就困难了。

回复
俊
2008年09月12日 15:28 编辑
大哥，不至于吧，我也得过脂肪瘤，但是处理起来比你方便多了，直接一刀开了。3天后就痊愈了……（话说也是背部）

回复
Coral
2008年09月12日 11:59 编辑
Bless you!
上班时间还能啰嗦这么多，不怕Sam炒鱿鱼阿！

回复
